src/adapter/src/error.rs

```diff
@@ -176,6 +176,8 @@ pub enum AdapterError {
     Unstructured(anyhow::Error),
     /// The named feature is not supported and will (probably) not be.
     Unsupported(&'static str),
+    /// The named feature is only enabled in unsafe mode.
+    RequiresUnsafe(&'static str),
```

Comment:
You shouldn't need this variant! Just return `AdapterError::PlanError(PlanError::RequiresUnsafe))` where you need it, no?

Reply:
I like that! We had separate code paths before and I was assuming that some of these would not be "plan errors", but it seems all call sites so far originate from parsing/processing SQL.